### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by edzh1**

### CODE REVIEW

File: lib/ahoy/ahoy.js [hunk 1]
```diff
@@ -82,8 +82,8 @@ export default class Ahoy {
   }
 
   initConnection = async () => {
-    this.hasInternetAccess = await NetInfo.isConnected.fetch();
-    NetInfo.isConnected.addEventListener('connectionChange', this.handleNetInfoConnectionChange);
+    let req = await fetch('https://www.google.com');
```
Comment (abaddonGIT) [inline lib/ahoy/ahoy.js]: I think change native connection change state on fetch to site - that a bad idea. if the site is not available, but this connection is active, we will get an error.

File: lib/ahoy/ahoy.js [hunk 2]
```diff
@@ -82,8 +82,8 @@ export default class Ahoy {
   }
 
   initConnection = async () => {
-    this.hasInternetAccess = await NetInfo.isConnected.fetch();
-    NetInfo.isConnected.addEventListener('connectionChange', this.handleNetInfoConnectionChange);
```
Comment (abaddonGIT) [inline lib/ahoy/ahoy.js]: I do not understand why deleted connection listener

Comment (edzh1) [reply lib/ahoy/ahoy.js]: hasInternetAccess is used only in 3 places: `addTrackingWorker`, `addVisitorWorker`, `createJob`.

`addTrackingWorker`, `addVisitorWorker` are called only once at start, so we don't need to watch for the internet connection for them.

`createJob` has this code:
`createJob = (name, event) => {
    if (this.offlineMode) this.hasInternetAccess = undefined;
    if (this.queue) this.queue.createJob(name, event);
  };`

And `hasInternetAccess` is useless here

File: lib/ahoy/ahoy.js (hunk 1 above)
Comment (edzh1) [reply lib/ahoy/ahoy.js]: I will use a native solution in a next PR

**Review by abaddonGIT — APPROVED** [review]

**Review by igor-slotin — APPROVED** [review]

**Pull request merged**